### PR TITLE
Read SLCS Granules from GUNW

### PR DESCRIPTION
**Pull request by bbuzz31**
Pulls the date/time directly from the SLC granules rather than the GUNW file name to address #488

### CODE REVIEW

File: tools/RAiDER/aria/calcGUNW.py [hunk 1]
```diff
@@ -88,8 +88,12 @@ def update_gunw_slc(path_gunw:str, ds_slc):
     with h5py.File(path_gunw, 'a') as h5:
         for k in TROPO_GROUP.split():
             h5 = h5[k]
-        del h5[TROPO_NAMES[0]]
-        del h5[TROPO_NAMES[1]]
+        # in case GUNW has already been updated once before
+        try:
+            del h5[TROPO_NAMES[0]]
+            del h5[TROPO_NAMES[1]]
+        except:
```
Comment (cmarshak) [inline tools/RAiDER/aria/calcGUNW.py:95]: Do not leave except statement without a specific error you expect. I think in this case you want KeyError.

File: tools/RAiDER/aria/prepFromGUNW.py [hunk 2]
```diff
@@ -13,6 +13,8 @@
 import yaml
 import shapely.wkt
 import RAiDER
+from dataclasses import dataclass
+
```
Comment (cmarshak) [inline tools/RAiDER/aria/prepFromGUNW.py:17]: Nice!

File: tools/RAiDER/losreader.py [hunk 3]
```diff
@@ -685,17 +686,21 @@ def getTopOfAtmosphere(xyz, look_vecs, toaheight, factor=None):
...
+def get_orbit(orbit_file, ref_time, pad):
     '''
     Returns state vectors from an orbit file
     '''
     # First load the state vectors into an isce orbit
     import isce3.ext.isce3 as isce
+    # svs  = get_sv(orbit_file, ref_time, pad)
+    # rows = np.stack([[isce.core.StateVector(row[0]), row[1:4], row[4:7]] for row in svs], axis=-1)
+    # orb  = isce.core.Orbit(rows)
+
```
Comment (cmarshak) [inline tools/RAiDER/losreader.py:698]: Please do not leave code that you used to debug in the code unless (a) you leave comments as to what you are doing to debug the code and (b) you and others will use this. When having to edit someone else's code and commented out code, the new editor has to think if this is purposefully commented out, when was this commented out and what this commented out code is debugging.

File: tools/RAiDER/models/gmao.py [hunk 4]
```diff
@@ -71,7 +71,7 @@ def _fetch(self, out):
         time1 = time
         time = round_time(time, 3 * 60 * 60)
         if not time1 == time:
-            logger.warning('Rounded given hour from  %d to %d. May be incorrect near beginning/end of day.', time1.hour, time.hour)
```
Comment (cmarshak) [inline tools/RAiDER/models/gmao.py:74]: Please use some more descriptive variable names like `original_time` and `rounded_time`?

**Review by cmarshak — CHANGES_REQUESTED** [review]
Hey Brett - I would normally leave the aesthetic stuff alone, but some of the patterns I really encourage you to fix. Will help maintain the longevity of this script particularly as others edit and debug including when it's in the cloud.

File: tools/RAiDER/aria/prepFromGUNW.py [hunk 5]
```diff
@@ -22,25 +24,24 @@
...
-        self.dates     = self.get_dates()
-        self.ref_time  = self.get_time()
-        self.look_dir  = self.get_look_dir()
+        self.dates, self.mid_time = self.get_datetimes()
+
+        self.look_dir   = self.get_look_dir()
         self.wavelength = self.get_wavelength()
-        self.name       = f'{self.dates[0]}-{self.dates[1]}'
-        self.OrbitFiles = self.get_orbit_files()
+        self.name       = self.make_fname()
```
Comment (dbekaert) [inline tools/RAiDER/aria/prepFromGUNW.py:40]: I assume this is only to make an intermediate file; does not propagate into the product?

Comment (bbuzz31) [reply tools/RAiDER/aria/prepFromGUNW.py:40]: Right, this is the name of the .cfg file which is used by RAiDER

**Review by cmarshak — APPROVED** [review]

**Comment by bbuzz31** [comment]
Screenshot 2023-03-07 at 16 39 38

**Comment by bbuzz31** [comment]
This is ready to be merged pending any last comments (sorry about the force;  rebased on the temporal interpolation)

@jlmaurer @cmarshak @dbekaert

File: tools/RAiDER/losreader.py [hunk 6]
```diff
@@ -231,6 +234,7 @@ def getLookVectors(self, ht, llh, xyz, yy):
                     los[ii, jj, :] = (sat_xyz - inp_xyz) / slant_range
                 except Exception as e:
                     los[ii, jj, :] = np.nan
+                    # print ('didnt get one')
```
Comment (jlmaurer) [inline tools/RAiDER/losreader.py:237]: This could be deleted

File: tools/RAiDER/losreader.py [hunk 7]
```diff
@@ -685,17 +686,20 @@ def getTopOfAtmosphere(xyz, look_vecs, toaheight, factor=None):
     return pos
 
 
-def get_orbit(orbit_file, ref_time, pad=600): 
+def get_orbit(orbit_file, ref_time, pad):
```
Comment (jlmaurer) [inline tools/RAiDER/losreader.py:689]: Is there a reason not to set a default? Seems like this function should have one.

File: tools/RAiDER/models/ecmwf.py [hunk 8]
```diff
@@ -163,9 +164,9 @@ def _get_from_ecmwf(self, lat_min, lat_max, lat_step, lon_min, lon_max,
 
         server = ecmwfapi.ECMWFDataServer()
 
-        corrected_date = util.round_date(time, datetime.timedelta(hours=6))
-        if not corrected_date == time:
-            logger.warning('Rounded given datetime from  %s to %s', time, corrected_date)
+        corrected_DT = util.round_date(time, datetime.timedelta(hours=self._time_res))
```
Comment (jlmaurer) [inline tools/RAiDER/models/ecmwf.py:167]: good catch!

**Review by jlmaurer — APPROVED** [review]
Just a couple minor things, otherwise LGTM!

File: tools/RAiDER/losreader.py (hunk 7 above)
Comment (bbuzz31) [reply tools/RAiDER/losreader.py:689]: The pad argument is passed several times throughout the code (i think 3) through higher level functions. This one is relatively low down so a higher one sets the padding. When debugging, I found it difficult to track where actually the padding value is getting set. So I set to just have one default value that's used at the top. I'm not sure what the best practice is but this seemed easiest for me.

**Pull request merged**